### PR TITLE
fix #10: allow unpinned local actions

### LINKED ISSUE #10: 1.1.0 release errors on local path actions, when 1.0.4 did not
there appears to be a regression since the last release. Now, this action is erroring on local actions. for example:

v 1.0.4 -> Correctly doesnt error for local actions
```
      - name: Ensure SHA pinned actions
        uses: zgosalvez/github-actions-ensure-sha-pinned-actions@8877889a5717dad0b139f1d2925689aa68f88a43
```
however, v1.1.0 -> Will error on my local action -> ```Error: ./.github/actions/prepare is not pinned to a full length commit SHA.```

```
      - name: Ensure SHA pinned actions
        uses: zgosalvez/github-actions-ensure-sha-pinned-actions@7692976b7529253c9610ff3242b2693d678847b0
```

Seems likely some unintended behavior was added in @xunleii recent merge

### PR DESCRIPTION
**Pull request by xunleii**
Try to fix regression issue (#10) after #7 when we use unpinned local actions

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -28,8 +28,8 @@ async function run() {
         const uses = jobs[job]['uses'];
         const steps = jobs[job]['steps'];
 
-        if (uses !== undefined) {
-          if (!assertUsesSHA(uses)) {
+        if (typeof uses === 'string') {
+          if (uses.includes('@') && !assertUsesSHA(uses)) {
```
Comment (zgosalvez) [inline src/index.js:32]:
```suggestion
          if (!assertUsesSHA(uses)) {
```

File: src/index.js [hunk 2]
```diff
@@ -39,7 +39,7 @@ async function run() {
           for (const step of steps) {
             const uses = step['uses'];
 
-            if (uses !== undefined && !assertUsesSHA(uses)) {
+            if (typeof uses === 'string' && uses.includes('@') && !assertUsesSHA(uses)) {
```
Comment (zgosalvez) [inline src/index.js:42]:
```suggestion
            if (!assertUsesSHA(uses)) {
```

**Review by zgosalvez — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 1 above)
Comment (xunleii) [reply src/index.js:32]: @zgosalvez On the `v1.0.4` release, this Github Action only ensure that GitHub Actions are pinned to full length commit SHAs. 
When I wrote my PR (#7), I forgot to only check a Github Action if it contains a `@`, forcing this validation for [local action](https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#example-using-an-action-in-the-same-repository-as-the-workflow) or a [Docker action](https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#example-using-a-docker-hub-action). So, checking if a `@` exists before checking if the reference is a SHA will fix the regression.

**Comment by zgosalvez** [comment]
I'll revisit this weekend, and I'll see if I can have some tests included to catch these use cases. Thank you for your patience.

**Comment by zgosalvez** [comment]
@xunleii, please check #12 and consider closing this PR.

**Pull request closed without merging**